### ISSUE TITLE
broken when in use with test packages

### ISSUE THREAD

**Issue by frioux** [mailru/easyjson#187]
If you use easyjson in the same directory as a package that also has test packages, it generates invalid code.  [Here's an example repo that has the problem.](https://github.com/frioux/easyjsontests/).  If you run `easyjson -pkg  from within the `a` directory of that repo you'll see the following error:

```
easyjson-bootstrap518172535.go:14:3: found packages a (a.go) and a_test (a_test_easyjson.go) in /home/frew/go/src/github.com/frioux/easyjsontests/a
Bootstrap failed: exit status 1
```

easyjson should probably make sure that what it generates is in the same package as the files with the special comments.

**Comment by aaron42net** [comment]
Perhaps easyjson should duplicate the logic from the Go compiler (https://github.com/golang/go/blob/master/src/go/build/build.go#L824) that if you are working in a file with a "_test.go" suffix and the discovered package name also has a suffix of "_test", strip that off.

**Comment by harshadptl** [comment]
Fixed it in parser. Whenever a package directory was parsed filewise, the package name in the recent file was updated as package name. 

This was creating problems while generating files.

https://github.com/mailru/easyjson/pull/201

**Comment by harshadptl** [comment]
Clean way to do this would have been to replicated what has been done here https://github.com/golang/go/blob/master/src/go/build/build.go#L824 . But filename is not available while 'Walking' ast nodes to check whether it is a test file.

**Comment by GoWebProd** [comment]
Fixed in #218
